javascript/i18n/phonenumbers/demo.js

```diff
@@ -50,71 +62,52 @@ function phoneNumberParser() {
     var isPossible = phoneUtil.isPossibleNumber(number);
     output.append('\nResult from isPossibleNumber(): ');
     output.append(isPossible);
-    if (!isPossible) {
+    var validationResult = i18n.phonenumbers.PhoneNumberUtil.ValidationResult;
+    var isPossibleReason = phoneUtil.isPossibleNumberWithReason(number)
+    if (isPossible) {
+      // Checking as isValid() fails if possible local only.
+      if (isPossibleReason == validationResult.IS_POSSIBLE_LOCAL_ONLY) {
+        output.append('\nResult from isPossibleNumberWithReason(): ');
+        output.append('IS_POSSIBLE_LOCAL_ONLY');
+        output.append(
+            '\nNumber is considered invalid as it is ' +
+            'not a possible national number.');
+      } else {
+        var isNumberValid = phoneUtil.isValidNumber(number);
+        output.append('\nResult from isValidNumber(): ');
+        output.append(isNumberValid);
+        if (isNumberValid && regionCode && regionCode != 'ZZ') {
```

Comment:
we could add the !ZZ check in the Java as well for consistency.

Reply:
Done. In java the !ZZ check is present, it is covered in boolean variable initialization hasRegionCode. Looks like they did for readability. Changed, as both the versions LGTM.

Reply:
Ahhhh thanks! I didn't go and check.
I'd go the other way and define hasDefaultRegion as a js var too, which is regionCode && regionCode != 'ZZ'.

Probably better.

Reply:
I also liked that apprach for readability :) 
Done.